diff --git a/qsym/pintool/expr.h b/qsym/pintool/expr.h
index 9539dcf..20f537e 100644
--- a/qsym/pintool/expr.h
+++ b/qsym/pintool/expr.h
@@ -811,7 +811,7 @@ protected:
   }
 
   z3::expr toZ3ExprRecursively(bool verbose) override {
-    return children_[0]->toZ3Expr(verbose) + children_[1]->toZ3Expr(verbose);
+    return children_[0]->toZ3Expr(verbose) - children_[1]->toZ3Expr(verbose);
   }
 };
 
