diff --git a/qsym/pintool/solver.cpp b/qsym/pintool/solver.cpp
index 921f506..7b17e92 100644
--- a/qsym/pintool/solver.cpp
+++ b/qsym/pintool/solver.cpp
@@ -236,15 +236,15 @@ void Solver::addJcc(ExprRef e, bool taken, ADDRINT pc) {
     if (debug_count == check_input_count) {
       if (debug_hash == check_input_hash) {
         if (debug_taken != check_input_taken) {
-          printf("Input is taking the expected direction!\n");
+          fprintf(stderr, "Input is taking the expected direction!\n");
           exit(0);
         } else {
-          printf("Input is divergent: it reaches the same branch but does not take the expected direction!\n");
+          fprintf(stderr, "Input is divergent: it reaches the same branch but does not take the expected direction!\n");
           if (debug_abort) abort();
           else exit(0);
         }
       } else {
-        printf("Input is divergent: it does take the same path! [hash is different: %x vs expected=%x]\n", debug_hash, check_input_hash);
+        fprintf(stderr, "Input is divergent: it does take the same path! [hash is different: %x vs expected=%x]\n", debug_hash, check_input_hash);
         if (debug_abort) abort();
         else exit(0);
       }
@@ -270,7 +270,7 @@ void Solver::addJcc(ExprRef e, bool taken, ADDRINT pc) {
 
   if (is_interesting)
     negatePath(e, taken);
-  addConstraint(e, taken, is_interesting);
+  addConstraint(e, /* taken */ true, is_interesting);
 
 #if DEBUG_CHECK_PI_SOLVER || DEBUG_CHECK_PI_CONCRETE
   static int check_pi = -1;
