diff --git a/compiler/Symbolizer.cpp b/compiler/Symbolizer.cpp
index 70843e1..15c7cff 100644
--- a/compiler/Symbolizer.cpp
+++ b/compiler/Symbolizer.cpp
@@ -857,7 +857,7 @@ void Symbolizer::visitCastInst(CastInst &I) {
 
     switch (I.getOpcode()) {
     case Instruction::SExt:
-      target = runtime.buildSExt;
+      target = runtime.buildZExt;
       break;
     case Instruction::ZExt:
       target = runtime.buildZExt;
